apps/davi/src/stores/types.ts

```diff
@@ -13,13 +15,15 @@ interface GovernanceCapabilities {
 // TODO: make a series of utils that parses the capabilities and translates them to a series of boolean flags, to make it easier to conditionally render UI elements
 
 type SupportedGovernanceSystem = 'SnapshotERC20Guild' | 'SnapshotRepGuild';
+type useGetNumberOfActiveProposalsUni =
+  | ReturnType<typeof useGetNumberOfActiveProposals>
+  | ReturnType<typeof useGetNumberOfActiveProposalsFromSubgraph>;
 
 // TODO: Wrap fetcher return types in a common FetcherHookReturn type which has common loading / error statuses
 export interface FetcherHooksInterface {
-  useGetActiveProposals: (daoId: string) => {
-    data: BigNumber;
-    refetch: () => void;
-  };
+  useGetNumberOfActiveProposals: (
+    daoId: string
+  ) => useGetNumberOfActiveProposalsUni;

```

Comment:
The previous types were OK. We might need to add `isLoading` and `isError` to track those if we need though.

We define our types here so we make sure that we're returning the same data, independently of the source or the governance implementation. That allows us to not care about it when consuming hook data in the components.

Reply:
Yes, `useProposal` is rather the exception. The way it's typed led to a rabbit hole of changed throughout the app. We should fix it at some point.

The rest of the hooks have the typings on the store because we don't want each hook implementation defining what will it return, but rather strictly define it globally in the store.

That allows us, from a component perspective, to be sure that each hook will return exactly the same information, and don't have to deal (from the component) with different data that depend either on the data source (subgraph or RPC) nor the governance implementation (guild, gov 1.5, snapshot, etc).

In the concrete case of `useGetNumberOfActiveProposals`, the component knows it'll receive a `BigNumber`. It's the hook that has to do the necessary transformations to ensure that.

If we let each hook return different types, the component will need to check if the hook returned a number, a string, or a BigNumber, and transform the data. That needs to be replicated each time we consume information from the hook.

Reply:
Thanks for this explanation! I understand that we want our store have the source of truth for our typings then from which our component will get it's typing from. Makes sense and I had missed this context. Thanks again, will make the changes necessary!